File: Problem-1-CheckEven/solution.js
// For Explanation read Solution.java

// Method 1
function usingBitwiseOper(n) {
    return (n & 1) == 0;
}

// Method 2
function usingIntegerDivision(n) {
    if((n / 2) * 2 == n) {
        return true;
    }
    return false;
}

console.log(usingBitwiseOper(0));
console.log(usingBitwiseOper(1));
console.log(usingBitwiseOper(4));
console.log(usingBitwiseOper(99));

// console.log(usingIntegerDivision(0));
// console.log(usingIntegerDivision(1));
// console.log(usingIntegerDivision(4));
// console.log(usingIntegerDivision(99));

File: Problem-1-CheckEven/solution.cpp
// For Explanation read Solution.java
#include<iostream>
using namespace std;

// Method 1
bool usingBitwiseOper(int n) {
    return (n & 1) == 0;
}

// Method 2
bool usingIntegerDivision(int n) {
    if((n / 2) * 2 == n) {
        return true;
    }
    return false;
}

int main() {
        
    cout << (usingBitwiseOper(0) ? "true" : "false") << endl;
    cout << (usingBitwiseOper(1) ? "true" : "false") << endl;
    cout << (usingBitwiseOper(4) ? "true" : "false") << endl;
    cout << (usingBitwiseOper(99) ? "true" : "false") << endl;

    // cout << usingIntegerDivision(0) ? "true" : "false" << endl;
    // cout << usingIntegerDivision(1) ? "true" : "false" << endl;
    // cout << usingIntegerDivision(4) ? "true" : "false" << endl;
    // cout << usingIntegerDivision(99) ? "true" : "false" << endl;

    return 0;
}
